src/hotspot/share/opto/vectorization.cpp

```diff
@@ -309,6 +309,27 @@ void VLoopDependencyGraph::compute_depth() {
     }
     set_depth(n, max_pred_depth + 1);
   }
+
+#ifdef ASSERT
+  for (int i = 0; i < _body.body().length(); i++) {
+    Node* n = _body.body().at(i);
+    int max_pred_depth = 0;
+    if (!n->is_Phi()) { // ignore backedge
+      for (PredsIterator it(*this, n); !it.done(); it.next()) {
+        Node* pred = it.current();
+        if (_vloop.in_bb(pred)) {
+          max_pred_depth = MAX2(max_pred_depth, depth(pred));
+        }
+      }
+    }
```

Comment:
This is identical to the loop above. Could this code be shared (e.g. `find_max_pred_depth()`).

Reply:
Good idea!